smack-experimental/src/main/java/org/jivesoftware/smackx/stanza_content_encryption/element/ContentElement.java

```diff
@@ -0,0 +1,291 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.stanza_content_encryption.element;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+
+import javax.xml.namespace.QName;
+
+import org.jivesoftware.smack.packet.ExtensionElement;
+import org.jivesoftware.smack.packet.XmlEnvironment;
+import org.jivesoftware.smack.util.Objects;
+import org.jivesoftware.smack.util.XmlStringBuilder;
+import org.jivesoftware.smackx.address.packet.MultipleAddresses;
+import org.jivesoftware.smackx.hints.element.MessageProcessingHint;
+import org.jivesoftware.smackx.sid.element.StanzaIdElement;
+
+import org.jxmpp.jid.Jid;
+
+/**
+ * Extension element that holds the payload element, as well as a list of affix elements.
+ * In SCE, the XML representation of this element is what will be encrypted using the encryption mechanism of choice.
+ */
+public class ContentElement implements ExtensionElement {
+
+    private static final String NAMESPACE_UNVERSIONED = "urn:xmpp:sce";
+    public static final String NAMESPACE_0 = NAMESPACE_UNVERSIONED + ":0";
+    public static final String NAMESPACE = NAMESPACE_0;
+    public static final String ELEMENT = "content";
+    public static final QName QNAME = new QName(NAMESPACE, ELEMENT);
+
+    private final PayloadElement payload;
+    private final List<AffixElement> affixElements;
+
+    ContentElement(PayloadElement payload, List<AffixElement> affixElements) {
+        this.payload = payload;
+        this.affixElements = Collections.unmodifiableList(affixElements);
+    }
+
+    /**
+     * Return the {@link PayloadElement} which holds the sensitive payload extensions.
+     *
+     * @return payload element
+     */
+    public PayloadElement getPayload() {
+        return payload;
+    }
+
+    /**
+     * Return a list of affix elements.
+     * Those are elements that need to be verified upon reception by the encryption mechanisms implementation.
+     *
+     * @see <a href="https://xmpp.org/extensions/xep-0420.html#affix_elements">
+     *     XEP-0420: Stanza Content Encryption - §4. Affix Elements</a>
+     *
+     * @return list of affix elements
+     */
+    public List<AffixElement> getAffixElements() {
+        return affixElements;
+    }
+
+    @Override
+    public String getNamespace() {
+        return NAMESPACE;
+    }
+
+    @Override
+    public String getElementName() {
+        return ELEMENT;
+    }
+
+    @Override
+    public CharSequence toXML(XmlEnvironment xmlEnvironment) {
+        XmlStringBuilder xml = new XmlStringBuilder(this).rightAngleBracket();
+        for (AffixElement affix : affixElements) {
```

Comment:
I think you can simply `xml.append(affixElements)`. If I am not mistaken, then there is a matching method in `XmlStringBuilder`.